Build breaks because of missing 'cuda.h' when ALUMINUM_ENABLE_CUDA=OFF

```
===>  Building for Aluminum-0.2.1
[1/39] /usr/local/bin/clang++70  -DAl_EXPORTS -I/wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/src -I. -isystem /usr/local/mpi/openmpi/include -isystem /usr/local/include -Wall -Wextra -pedantic -g -faligned-new -O2 -pipe -fstack-protector -fno-strict-aliasing -fPIC   -pthread -fopenmp=libomp -std=gnu++11 -MD -MT src/CMakeFiles/Al.dir/profiling.cpp.o -MF src/CMakeFiles/Al.dir/profiling.cpp.o.d -o src/CMakeFiles/Al.dir/profiling.cpp.o -c /wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/src/profiling.cpp
[2/39] /usr/local/bin/clang++70   -I/wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/test -I/wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/src -I. -isystem /usr/local/mpi/openmpi/include -isystem /usr/local/include -Wall -Wextra -pedantic -g -faligned-new -O2 -pipe -fstack-protector -fno-strict-aliasing -fPIE   -pthread -fopenmp=libomp -std=gnu++11 -MD -MT benchmark/CMakeFiles/benchmark_waits.exe.dir/benchmark_waits.cpp.o -MF benchmark/CMakeFiles/benchmark_waits.exe.dir/benchmark_waits.cpp.o.d -o benchmark/CMakeFiles/benchmark_waits.exe.dir/benchmark_waits.cpp.o -c /wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/benchmark/benchmark_waits.cpp
FAILED: benchmark/CMakeFiles/benchmark_waits.exe.dir/benchmark_waits.cpp.o 
/usr/local/bin/clang++70   -I/wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/test -I/wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/src -I. -isystem /usr/local/mpi/openmpi/include -isystem /usr/local/include -Wall -Wextra -pedantic -g -faligned-new -O2 -pipe -fstack-protector -fno-strict-aliasing -fPIE   -pthread -fopenmp=libomp -std=gnu++11 -MD -MT benchmark/CMakeFiles/benchmark_waits.exe.dir/benchmark_waits.cpp.o -MF benchmark/CMakeFiles/benchmark_waits.exe.dir/benchmark_waits.cpp.o.d -o benchmark/CMakeFiles/benchmark_waits.exe.dir/benchmark_waits.cpp.o -c /wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/benchmark/benchmark_waits.cpp
/wrkdirs/usr/ports/net/aluminum/work/Aluminum-0.2.1/benchmark/benchmark_waits.cpp:4:10: fatal error: 'cuda.h' file not found
#include <cuda.h>
         ^~~~~~~~
1 error generated.
```

--- a/benchmark/CMakeLists.txt
+++ b/benchmark/CMakeLists.txt
@@ -31,18 +31,16 @@ if (AL_HAS_CUDA)
 endif ()
 target_include_directories(benchmark_pingpong.exe PRIVATE ${CMAKE_SOURCE_DIR}/test)
 
-add_executable(benchmark_events.exe benchmark_events.cpp
-  $<$<BOOL:${AL_HAS_CUDA}>:wait.cu>)
-target_link_libraries(benchmark_events.exe PRIVATE Al aluminum_test_headers)
 if (AL_HAS_CUDA)
+  add_executable(benchmark_events.exe benchmark_events.cpp wait.cu)
+  target_link_libraries(benchmark_events.exe PRIVATE Al aluminum_test_headers)
   target_link_libraries(benchmark_events.exe PUBLIC cuda)
-endif ()
-target_include_directories(benchmark_events.exe PRIVATE ${CMAKE_SOURCE_DIR}/test)
+  target_include_directories(
+    benchmark_events.exe PRIVATE ${CMAKE_SOURCE_DIR}/test)
 
-add_executable(benchmark_waits.exe benchmark_waits.cpp
-  $<$<BOOL:${AL_HAS_CUDA}>:wait.cu>)
-target_link_libraries(benchmark_waits.exe PRIVATE Al aluminum_test_headers)
-if (AL_HAS_CUDA)
+  add_executable(benchmark_waits.exe benchmark_waits.cpp wait.cu)
+  target_link_libraries(benchmark_waits.exe PRIVATE Al aluminum_test_headers)
   target_link_libraries(benchmark_waits.exe PUBLIC cuda)
+  target_include_directories(
+    benchmark_waits.exe PRIVATE ${CMAKE_SOURCE_DIR}/test)
 endif ()
-target_include_directories(benchmark_waits.exe PRIVATE ${CMAKE_SOURCE_DIR}/test)

--- a/benchmark/benchmark_pingpong.cpp
+++ b/benchmark/benchmark_pingpong.cpp
@@ -132,6 +132,8 @@ int main(int argc, char** argv) {
   do_benchmark();
   Al::Finalize();
 #else
+  (void) argc;
+  (void) argv;
   std::cout << "MPI-CUDA support required" << std::endl;
 #endif
   return 0;

--- a/test/test_multi_nballreduces.cpp
+++ b/test/test_multi_nballreduces.cpp
@@ -86,6 +86,7 @@ void test_multiple_nballreduces() {
         Al::Allreduce<Al::MPIBackend>(blocking_data.data(), blocking_data.size(),
                                       Al::ReductionOperator::sum, comm);
       }*/
+      (void) num_blocking;
       // Complete and check them.
       for (size_t i = 0; i < num_concurrent; ++i) {
         Al::Wait<Backend>(reqs[i]);

